mcs/mcs/expression.cs

```diff
@@ -11890,6 +11891,11 @@ public override void Emit (EmitContext ec)
 			int size = BuiltinTypeSpec.GetSize (otype);
 
 			count.Emit (ec);
+			bool count_on_stack = false;
+			if (ctor != null && !ExpressionAnalyzer.IsInexpensiveLoad (count)) {
+				ec.Emit (OpCodes.Dup);
+				count_on_stack = true;
```

Comment:
why not always `dup`?

Reply:
Because our jit cannot decompose dup in most cases and it feels odd to me to emit dup for constant values